libraries/chain/include/graphene/chain/proposal_evaluator.hpp

```diff
@@ -68,4 +68,67 @@ namespace graphene { namespace chain {
          const proposal_object* _proposal = nullptr;
    };
 
+   namespace impl {
```

Comment:
I'd put impl code in cpp file.

Reply:
i am not following here, the visitor class code in a cpp ? don't make sense. pls explain a bit more. thanks :)

Reply:
IMHO implementation related code should be in cpp file, since usually it's no need to be exposed to other classes. Not sure why @pmconrad put it in a hpp file in the first place. If you check the code base, most of visitor classes are defined in cpp files, except the ones used in templates that have to be put in hpp files. @pmconrad what's your opinion?

Reply:
I don't remember why I put it in there.
I agree with @abitmore that implementation-specific stuff doesn't belong in .hpp files.